managed/services/server/server.go

```diff
@@ -550,17 +550,30 @@ func (s *Server) validateChangeSettingsRequest(ctx context.Context, req *serverp
 		return status.Error(codes.FailedPrecondition, "DBaaS is enabled via ENABLE_DBAAS or via deprecated PERCONA_TEST_DBAAS environment variable.")
 	}
 
-	if metricsRes.GetHr().AsDuration() != 0 && s.envSettings.MetricsResolutions.HR != 0 {
+	if metricsRes.GetHr().AsDuration() != 0 &&
```

Comment:
Can we use a separate function to capture conditions here? Same comment for following modifications too.

Reply:
good idea